analyzer/runtime/evm/erc165.go

```diff
@@ -0,0 +1,79 @@
+package evm
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/oasisprotocol/oasis-core/go/common/errors"
+
+	"github.com/oasisprotocol/oasis-indexer/analyzer/evmabi"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
+)
+
+func InterfaceID(abi *abi.ABI) []byte {
+	id := []byte{0, 0, 0, 0}
+	for _, m := range abi.Methods {
+		id[0] ^= m.ID[0]
+		id[1] ^= m.ID[1]
+		id[2] ^= m.ID[2]
+		id[3] ^= m.ID[3]
+	}
+	return id
+}
+
+var (
+	InvalidInterfaceID             = []byte{0xff, 0xff, 0xff, 0xff}
+	ERC165InterfaceID              = InterfaceID(evmabi.ERC165)
+	ERC721InterfaceID              = InterfaceID(evmabi.ERC721)
+	ERC721TokenReceiverInterfaceID = InterfaceID(evmabi.ERC721TokenReceiver)
+	ERC721MetadataInterfaceID      = InterfaceID(evmabi.ERC721Metadata)
+	ERC721EnumerableInterfaceID    = InterfaceID(evmabi.ERC721Enumerable)
+)
+
+const ERC165GasLimit uint64 = 30_000
```

Comment:
why do we need this instead of `DefaultGasLimit`?

Reply:
erc-165 says they must work with this amount of gas and puts forth an algorithm in which you use this amount of gas. the default gas used in other calls is a guess that could theoretically be even lower than this, which could cause us to miss things

Reply:
keeping this. marking as resolved